src/coreComponents/fileIO/python/PyVTKOutput.cpp

```diff
@@ -0,0 +1,204 @@
+
+  VERIFY_INITIALIZED( pyVTKOutput );
+
+  string repr;
+  string const path = pyVTKOutput->group->getPath();
+  string const type = LvArray::system::demangle( typeid( *(pyVTKOutput->group) ).name() );
+  repr = path + " ( " + type + " )";
+
+  return PyUnicode_FromString( repr.c_str() );
+
+}
+
+
+static PyObject * output( PyVTKOutput * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  double time;
+  double dt;
+
+  if( !PyArg_ParseTuple( args, "dd", &time, &dt ) )
+  {
+    return nullptr;
+  }
+
+  geosx::ProblemManager & problem = self->group->getGroupByPath< ProblemManager >( "/Problem/" );
+  geosx::DomainPartition & domain = problem.getDomainPartition();
+
+  int cycleNumber = int(time/dt);
+  try
+  {
+    self->group->execute( time, dt, cycleNumber, 0, 0, domain );
+  }
+  catch( std::out_of_range const & e )
+  {
+    std::cout<<"Target not found. Impossible output."<<std::endl;
+  }
+  Py_RETURN_NONE;
+}
+
+static PyObject * setOutputDir( PyVTKOutput * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  PyObject * unicodePath;
+  // string fileNameRoot;
+
+  if( !PyArg_ParseTuple( args, "U", &unicodePath ) )
+  {
+    return nullptr;
+  }
+
+  LvArray::python::PyObjectRef<> asciiPath { PyUnicode_AsASCIIString( unicodePath ) };
+  if( asciiPath == nullptr )
+  {
+    return nullptr;
+  }
+
+  char const * const path = PyBytes_AsString( asciiPath );
+  if( path == nullptr )
+  {
+    return nullptr;
+  }
+
+  try
+  {
+    self->group->setOutputDirectory( path );
+    self->group->postProcessInput();
+  }
+  catch( std::out_of_range const & e )
+  {
+    std::cout<<"Target not found."<<std::endl;
+  }
+  Py_RETURN_NONE;
+}
+
+static PyObject * setOutputFileRootName( PyVTKOutput * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  PyObject * fileNameRoot;
+
+  if( !PyArg_ParseTuple( args, "U", &fileNameRoot ) )
+  {
+    return nullptr;
+  }
+
+  LvArray::python::PyObjectRef<> asciiFileNameRoot { PyUnicode_AsASCIIString( fileNameRoot ) };
+  if( asciiFileNameRoot == nullptr )
+  {
+    return nullptr;
+  }
+
+  char const * const filename = PyBytes_AsString( asciiFileNameRoot );
+  if( filename == nullptr )
+  {
+    return nullptr;
+  }
+
+  try
+  {
+    self->group->setPlotFileRoot( filename );
+    self->group->postProcessInput();
+  }
+  catch( std::out_of_range const & e )
+  {
+    std::cout<<"Target not found."<<std::endl;
+  }
+  Py_RETURN_NONE;
+}
+
+
+static PyMethodDef PyVTKOutput_methods[] = {
+  { "output", (PyCFunction) output, METH_VARARGS, "wrapper to routine VTKOutput::execute"},
+  { "setOutputDir", (PyCFunction) setOutputDir, METH_VARARGS, "wrapper to change directory output"},
+  { "setOutputFileRootName", (PyCFunction) setOutputFileRootName, METH_VARARGS, "wrapper to change the root name of the output files and subfolders"},
+  { nullptr, nullptr, 0, nullptr }      /* Sentinel */
```

Comment:
I would add a `reinit()` method that calls the GEOSX `reinit()` method which would be overwriten in the `VtkFiles` with a call to `postProcessInput()` probably.

Reply:
Edit : Since the `postProcessInput()` call in the `set()` routines works. There is no need of `reinit()` here.